Sources/PushNotifications.swift

```diff
@@ -89,6 +96,54 @@ import Foundation
     }
     #endif
 
+    /**
+     Set user id.
+
+     - Parameter userId: User id.
+
+     - Precondition: You need to first authenticate the user by providing `beamsTokenProvider` in `start(instanceId:, beamsTokenProvider:)` method. `beamsTokenProvider` should not be nil.
+    */
+    /// - Tag: setUserId
+    @objc public func setUserId(_ userId: String) throws {
+        guard let beamsTokenProvider = self.beamsTokenProvider else {
+            throw UserValidationtError.beamsTokenProviderException
+        }
+
+        let persistenceService: UserPersistable = PersistenceService(service: UserDefaults(suiteName: "PushNotifications")!)
+        guard persistenceService.getUserId() == nil else {
+            throw UserValidationtError.userExists
+        }
+
+        beamsTokenProvider.fetchToken(userId: userId, completion: { (token) in
+
+            guard
+                let deviceId = Device.getDeviceId(),
+                let instanceId = Instance.getInstanceId(),
+                let url = URL(string: "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances/\(instanceId)/devices/apns/\(deviceId)/user")
+                else {
+                    return
```

Comment:
Returning without any indication of what the error was could make debugging things tricky!